### PR TITLE
Initial tokentpr spec

### PR DESCRIPTION
**Pull request by nhlfr**
tokentpr will be used by token-operator as a request to generate kubeadm token for the specified cluster. It contains only the cluster ID and doens't need more information.

Ref giantswarm/k8scloudconfig#42

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright {yyyy} {name of copyright owner}
+   Copyright 2016 Giant Swarm GmbH
```
Comment (rossf7) [inline LICENSE]: Nitpick and not 100% but I think this should be 2017

File: glide.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package: github.com/giantswarm/tokentpr
+import:
+- package: k8s.io/client-go
+  version: ~2.0.0-alpha.0
```
Comment (rossf7) [inline glide.yaml]: Why not use ~2.0.0?

File: meta.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package tokentpr
+
+// Metadata to create operatorkit TPR.
+
+const (
+	// Name represents the name of the third party resource within Kubernetes.
+	Name = "token.giantswarm.io"
```
Comment (xh3b4sd) [inline meta.go]: I think `token` is a very general name. What about `kube-adm-token` or something more specific?

Comment (nhlfr) [reply meta.go]: sounds good

Comment (xh3b4sd) [reply meta.go]: Then lets please also rename the repo. Shouldn't be too bad in this early stage.

File: LICENSE (hunk 1 above)
Comment (nhlfr) [reply LICENSE]: done

File: glide.yaml (hunk 2 above)
Comment (nhlfr) [reply glide.yaml]: done

File: meta.go (hunk 3 above)
Comment (nhlfr) [reply meta.go]: done

File: custom_object.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package kubeadmtokentpr
+
+import (
+	"k8s.io/client-go/pkg/api/unversioned"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+// CustomObject represents the Flannel TPR's custom object. It holds the
```
Comment (rossf7) [inline custom_object.go]: GoDoc needs to be updated.

**Review by rossf7 — COMMENTED** [review]
@nhlfr found some out of date godoc but then I think this is good to go :)

File: custom_object.go (hunk 4 above)
Comment (nhlfr) [reply custom_object.go]: whoops

Comment (nhlfr) [reply custom_object.go]: But the worst thing is that I copied that from ingresstpr

Comment (nhlfr) [reply custom_object.go]: done

File: custom_object.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package kubeadmtokentpr
+
+import (
+	"k8s.io/client-go/pkg/api/unversioned"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+// CustomObject represents the token TPR's custom object. It holds the
+// specifications of the resource the Flannel operator is interested in.
```
Comment (rossf7) [inline custom_object.go]: Another ref to flannel

**Review by rossf7 — APPROVED** [review]
Found another ref to Flannel. Pls fix that and then LGTM :)

File: custom_object.go (hunk 4 above)
Comment (xh3b4sd) [reply custom_object.go]: :D

**Pull request merged**